### PR TITLE
build: refine build and lint

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
refine build and lint

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cmd/dm-syncer/config.go [hunk 1]
```diff
@@ -221,14 +218,14 @@ type syncerConfig struct {
...
-	ServerID int    `toml:"server-id" json:"server-id"`
-	Meta     string `toml:"meta" json:"meta"`
-	// NOTE: This item is deprecated.
+	StatusAddr         string `toml:"status-addr" json:"status-addr"`
+	Meta               string `toml:"meta" json:"meta"`
 	PersistentTableDir string `toml:"persistent-dir" json:"persistent-dir"`
 	Flavor             string `toml:"flavor" json:"flavor"`
 
+	ServerID int `toml:"server-id" json:"server-id"`
+	// NOTE: This item is deprecated.
```
Comment (lance6716) [inline cmd/dm-syncer/config.go:227]: comment should be put above the item 🤔

**Review by lance6716 — APPROVED** [review]
too long 😢

**Comment by GMHDBJD** [comment]
May test lint error can be detected in ci later. WIP. 😂
/hold

**Comment by GMHDBJD** [comment]
> too long 😢

Relay on test

**Comment by GMHDBJD** [comment]
PTAL
/unhold

File: .github/workflows/check-and-build.yml [hunk 2]
```diff
@@ -56,11 +56,5 @@ jobs:
...
         with:
-          version: v1.31
-          args: --timeout 10m0s --skip-dirs ^_tool/
-
-      - name: Revive Lint
-        uses: morphy2k/revive-action@v1
-        with:
-          config: .revive.toml
-          exclude: _tool/...
+          version: v1.39
```
Comment (lance6716) [inline .github/workflows/check-and-build.yml:59]: why not latest

**Comment by GMHDBJD** [comment]
DM-117 recorded
/run-all-tests

File: Makefile [hunk 3]
```diff
@@ -128,33 +122,26 @@ unit_test_pkg_binlog: tools_setup
...
 lint: tools_setup
-	@echo "golint"
-	tools/bin/golint -set_exit_status $(PACKAGES)
-
-vet:
-	$(GO) build -o bin/shadow golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
-	@echo "vet"
-	@$(GO) vet -composites=false $(PACKAGES)
-	@$(GO) vet -vettool=$(CURDIR)/bin/shadow $(PACKAGES) || true
+	@if [[ "${nolint}" != "true" ]]; then\
```
Comment (lance6716) [inline Makefile:132]: how did we use it

File: dm/config/source_config.go [hunk 4]
```diff
@@ -34,17 +35,18 @@ var getAllServerIDFunc = utils.GetAllServerID
...
+// PurgeConfig is the configuration for Purger.
 type PurgeConfig struct {
 	Interval    int64 `yaml:"interval" toml:"interval" json:"interval"`             // check whether need to purge at this @Interval (seconds)
 	Expires     int64 `yaml:"expires" toml:"expires" json:"expires"`                // if file's modified time is older than @Expires (hours), then it can be purged
 	RemainSpace int64 `yaml:"remain-space" toml:"remain-space" json:"remain-space"` // if remain space in @RelayBaseDir less than @RemainSpace (GB), then it can be purged
 }
 
-// SourceConfig is the configuration for Worker
+// SourceConfig is the configuration for Worker.
+// SourceConfig is the configuration for Worker.
```
Comment (lance6716) [inline dm/config/source_config.go:49]:
```suggestion
```

File: dm/config/subtask.go [hunk 5]
```diff
@@ -264,9 +259,9 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		return terror.ErrConfigTooLongSourceID.Generate()
 	}
 
-	//if c.Flavor != mysql.MySQLFlavor && c.Flavor != mysql.MariaDBFlavor {
+	// if c.Flavor != mysql.MySQLFlavor && c.Flavor != mysql.MariaDBFlavor {
```
Comment (lance6716) [inline dm/config/subtask.go:262]: we could delete it now 🤔

File: dm/ctl/common/util.go [hunk 6]
```diff
@@ -23,12 +23,13 @@ import (
...
+	"go.etcd.io/etcd/clientv3"
+
 	"github.com/pingcap/dm/dm/config"
 	"github.com/pingcap/dm/dm/pb"
 	parserpkg "github.com/pingcap/dm/pkg/parser"
 	"github.com/pingcap/dm/pkg/terror"
 	"github.com/pingcap/dm/pkg/utils"
-	"go.etcd.io/etcd/clientv3"
 
 	"github.com/golang/protobuf/jsonpb"
```
Comment (lance6716) [inline dm/ctl/common/util.go:34]: don't know why CI didn't group all 3-rd package together

**Review by lance6716 — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (GMHDBJD) [reply Makefile:132]: In check-and-build.yml. Since we already have a lint step, don't need to lint in build step

**Comment by lance6716** [comment]
(and I hope the next PR to be merged is https://github.com/pingcap/dm/pull/1412

File: dm/worker/server_test.go [hunk 7]
```diff
@@ -494,12 +495,12 @@ func (t *testServer) testOperateWorker(c *C, s *Server, dir string, start bool)
 			currentWorker := s.getWorker(true)
 			return currentWorker == nil && w.closed.Get()
 		}), IsTrue)
-		c.Assert(s.getSourceStatus(true).Result, IsNil)
+		c.Assert(s.getSourceStatus(false).Result, IsNil)
```
Comment (lance6716) [inline dm/worker/server_test.go:498]:
```suggestion
		c.Assert(s.getSourceStatus(true).Result, IsNil)
```

data race

**Pull request merged**